indexing-service/src/main/java/org/apache/druid/indexing/firehose/IngestSegmentFirehoseFactory.java

```diff
@@ -204,87 +204,87 @@ public Firehose connect(InputRowParser inputRowParser, File temporaryDirectory)
         segmentIds
     );
 
-    try {
-      final List<TimelineObjectHolder<String, DataSegment>> timeLineSegments = getTimeline();
-
-      // Download all segments locally.
-      // Note: this requires enough local storage space to fit all of the segments, even though
-      // IngestSegmentFirehose iterates over the segments in series. We may want to change this
-      // to download files lazily, perhaps sharing code with PrefetchableTextFilesFirehoseFactory.
-      final SegmentLoader segmentLoader = segmentLoaderFactory.manufacturate(temporaryDirectory);
-      Map<DataSegment, File> segmentFileMap = Maps.newLinkedHashMap();
-      for (TimelineObjectHolder<String, DataSegment> holder : timeLineSegments) {
-        for (PartitionChunk<DataSegment> chunk : holder.getObject()) {
-          final DataSegment segment = chunk.getObject();
-          if (!segmentFileMap.containsKey(segment)) {
-            segmentFileMap.put(segment, segmentLoader.getSegmentFiles(segment));
+    final List<TimelineObjectHolder<String, DataSegment>> timeLineSegments = getTimeline();
+
+    // Download all segments locally.
+    // Note: this requires enough local storage space to fit all of the segments, even though
+    // IngestSegmentFirehose iterates over the segments in series. We may want to change this
+    // to download files lazily, perhaps sharing code with PrefetchableTextFilesFirehoseFactory.
+    final SegmentLoader segmentLoader = segmentLoaderFactory.manufacturate(temporaryDirectory);
+    Map<DataSegment, File> segmentFileMap = Maps.newLinkedHashMap();
+    for (TimelineObjectHolder<String, DataSegment> holder : timeLineSegments) {
+      for (PartitionChunk<DataSegment> chunk : holder.getObject()) {
+        final DataSegment segment = chunk.getObject();
+
+        segmentFileMap.computeIfAbsent(segment, k -> {
+          try {
+            return segmentLoader.getSegmentFiles(segment);
```

Comment:
segmentLoader.getSegmentFiles(segment) throws SegmentLoadingException, which needs to be caught and re thrown as RuntimeException. This makes the outer try catch redundant (compilation error) as the SegmentLoadingException has already been caught.